packages/grafana-ui/src/components/TimePicker/TimePicker.tsx

```diff
@@ -183,6 +183,7 @@ class UnThemedTimePicker extends PureComponent<Props, State> {
       </>
     );
     const isAbsolute = isDateTime(value.raw.to);
+    const isLongDate = typeof this.props.value.raw.from !== 'string' || typeof this.props.value.raw.to !== 'string';
```

Comment:
Wouldnt `isAbsolute` be enough?

Reply:
@davkal Sure, though we are going to have small inconsistencies then.

> 2019-10-01 00:00:00 to a few seconds ago
aka **date-to-now** would be a short date (with isAbsolute).

> 5 minutes ago to 2019-10-09 23:59:59
aka **now-minus-offset-to-date** would be a long date, even though technically it's shorter than date-to-now.

With **isLongDate** both of them are classified as long dates.

Reply:
@davkal Sure, though we are going to have small inconsistencies then.

With isAbsolute:

> 2019-10-01 00:00:00 to a few seconds ago

aka **date-to-now** would be classified as a short date.

> 5 minutes ago to 2019-10-09 23:59:59

aka **now-minus-offset-to-date** would be classified a long date, even though technically it's shorter than date-to-now in an example above.

With **isLongDate** both of them are classified as long dates.

Reply:
I think for the purpose of collapsing the button, it's fine if either is already an absolute date, perhaps `isAbsolute` could change to `hasAbsolute` if that works with the rest of the code. 

I guess the biggest issue I have is that we're doing 2 different kinds of detection here: one with `typeof` and the other one with `isDateTime` just above. Can we use either and not both? If we need to use both, then I'd expect a comment.

Reply:
@davkal of course, we don't need them both. I already fixed that - renamed isAbsolute to hasAbsolute and consolidated long datetime detection within hasAbsolute.

```ts
const hasAbsolute = isDateTime(value.raw.from) || isDateTime(value.raw.to);
```

Reply:
Happy with this.